File: frontend/src/types/HealthWellbeingRecord.ts
export interface HealthWellbeingRecord {
  healthRecordId: number;
  residentId: number;
  recordDate: string;
  generalHealthScore?: number;
  nutritionScore?: number;
  sleepQualityScore?: number;
  energyLevelScore?: number;
  heightCm?: number;
  weightKg?: number;
  bmi?: number;
  medicalCheckupDone?: boolean;
  dentalCheckupDone?: boolean;
  psychologicalCheckupDone?: boolean;
  notes?: string;
}

File: backend/Beacon.API/Models/HealthWellbeingRecord.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Beacon.API.Models;

public partial class HealthWellbeingRecord
{
    [Key]
    public int HealthRecordId { get; set; }

    public int ResidentId { get; set; }

    public DateOnly RecordDate { get; set; }

    public decimal? GeneralHealthScore { get; set; }

    public decimal? NutritionScore { get; set; }

    public decimal? SleepQualityScore { get; set; }

    public decimal? EnergyLevelScore { get; set; }

    public decimal? HeightCm { get; set; }

    public decimal? WeightKg { get; set; }

    public decimal? Bmi { get; set; }

    public bool? MedicalCheckupDone { get; set; }

    public bool? DentalCheckupDone { get; set; }

    public bool? PsychologicalCheckupDone { get; set; }

    public string? Notes { get; set; }

    public virtual Resident Resident { get; set; } = null!;
}
